src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -421,31 +411,35 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
         return kernel;
     }
 
-    private async checkForIpykernel6(doc: NotebookDocument, waitingMessage?: string): Promise<boolean> {
+    private async checkForIpykernel6(doc: NotebookDocument): Promise<boolean> {
         try {
-            const controller = this.notebookControllerManager.getSelectedNotebookController(doc);
-            const interpreter = controller?.connection.interpreter;
-            if (interpreter) {
-                const cacheResult = this.cache.get(interpreter);
-                if (cacheResult === true) {
-                    return true;
+            let kernel = this.kernelProvider.get(doc);
+
+            if (!kernel) {
+                const controller = this.notebookControllerManager.getSelectedNotebookController(doc);
+
+                if (controller) {
+                    kernel = this.kernelProvider.getOrCreate(doc, {
+                        metadata: controller.connection,
+                        controller: controller?.controller,
+                        resourceUri: doc.uri
+                    });
                 }
+            }
 
-                const checkCompatible = () =>
-                    this.pythonInstaller.isProductVersionCompatible(Product.ipykernel, '>=6.0.0', interpreter);
-                const status = waitingMessage
-                    ? await this.appShell.withProgress(
-                          { location: ProgressLocation.Notification, title: waitingMessage },
-                          checkCompatible
-                      )
-                    : await checkCompatible();
-                const result = status === ProductInstallStatus.Installed;
-
-                sendTelemetryEvent(DebuggingTelemetry.ipykernel6Status, undefined, {
-                    status: result ? 'installed' : 'notInstalled'
-                });
-                this.cache.set(interpreter, result);
-                return result;
+            if (kernel) {
+                const code = 'import ipykernel\nprint(ipykernel.__version__)';
+                const output = await kernel.executeHidden(code);
+
+                if (output[0].text) {
+                    const majorVersion = Number(output[0].text.toString().charAt(0));
+                    const result = majorVersion >= 6;
```

Comment:
Will work until ipykernel gets to version 10

Reply:
OMG, thanks

Reply:
addressed in https://github.com/microsoft/vscode-jupyter/pull/7695